### PR TITLE
[DMs] Add styles to DMs composer, scrolling fixes

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

- Fixes reactions by sending chat id and updating SDK
- Fixes the reaction store to always have the same kind of ID
- Fixes the layout to allow scrolling under the header
- Fixes messages to preserve whitespace
- Adds a new `TextArea` component that grows, is more customizable and doesn't rely on absolute positioning
- Adds the Figma styles (sans correct icon) to the message composer for chats
- Automatically scroll to bottom of the messages if there's a new message and the user was at the bottom before it came in
- Scroll to the bottom of the existing messages window if the user clicks on the active chat in the sidebar
- Fixes react key error on separator/chat messages

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-composer

File: packages/web/src/components/data-entry/TextAreaV2.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import { useRef, useCallback, useEffect, ComponentPropsWithoutRef } from 'react'
```
Comment (rickyrombo) [inline packages/web/src/components/data-entry/TextAreaV2.tsx:1]: @raymondjacobson / @dylanjeffers should we add TextArea and other inputs to stems?

File: packages/web/src/pages/chat-page/ChatPage.tsx [hunk 2]
```diff
@@ -34,15 +92,31 @@ export const ChatPage = ({ match }: RouteComponentProps<{ id?: string }>) => {
...
+          <ChatList
+            className={styles.chatList}
+            currentChatId={currentChatId}
+            onChatClicked={handleChatClicked}
+          />
         </div>
-        <div className={styles.messages}>
-          <ChatMessageList className={styles.messageList} chatId={chatId} />
-          <ChatComposer className={styles.messageComposer} chatId={chatId} />
+        <div className={styles.chatArea}>
```
Comment (rickyrombo) [inline packages/web/src/pages/chat-page/ChatPage.tsx:105]: Doing the scrolling stuff was interesting. I chose using this element as the scroll area to make having the composer stick to the bottom a lot easier. The side effect is I have to move it up to slide it under the header for that nice blur effect. I can do either way, but this seemed the cleanest

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-composer

File: packages/web/src/components/data-entry/TextAreaV2.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  }, [grows, growTextArea, value])
+
+  const style = {
+    [styles.noResize]: !resize,
+    [styles.medium]: size === TextAreaSize.MEDIUM,
+    [styles.small]: size === TextAreaSize.SMALL
+  }
+
+  const nearCharacterLimit = maxLength
+    ? `${value}`.length > (7.0 / 8.0) * maxLength
```
Comment (piazzatron) [inline packages/web/src/components/data-entry/TextAreaV2.tsx:71]: nit: maybe we pull this out?

File: packages/web/src/pages/chat-page/ChatPage.tsx [hunk 4]
```diff
@@ -1,28 +1,86 @@
...
+  const messagesRef = useRef<HTMLDivElement>(null)
+
+  // Cache whether the user was already scrolled to the bottom
+  // Effects run after the render, so we can't calculate this inside the effect
+  // but without rendering, there's nowhere to scroll, so do that within the effect
+  let wasAtBottom = false
+  if (messagesRef.current) {
+    const { scrollTop, clientHeight, scrollHeight } = messagesRef.current
+    wasAtBottom = scrollTop + clientHeight >= scrollHeight
+  }
```
Comment (piazzatron) [inline packages/web/src/pages/chat-page/ChatPage.tsx:57]: maybe pull this out into a standalone hook?

File: packages/web/src/pages/chat-page/components/ChatComposer.tsx [hunk 5]
```diff
@@ -6,46 +6,81 @@ import {
...
-          type='text'
-          value={message}
-          onChange={handleInputChange}
-        />
+          placeholder={messages.sendMessagePlaceholder}
+          onKeyDown={handleKeyDown}
+          onChange={handleChange}
+          value={value}
+          maxVisibleRows={3}
+          maxLength={500}
```
Comment (piazzatron) [inline packages/web/src/pages/chat-page/components/ChatComposer.tsx:73]: let's note to ask product what a sensible max length here is

File: packages/web/src/pages/chat-page/components/ChatMessageList.tsx [hunk 6]
```diff
@@ -60,23 +60,22 @@ export const ChatMessageList = (props: ChatMessageListProps) => {
...
     }
-    const pollInterval = setInterval(() => {
-      if (chatId) {
-        dispatch(fetchNewChatMessages({ chatId }))
-      }
-    }, 100000000000000000)
-    return () => {
-      clearInterval(pollInterval)
-    }
+    // const pollInterval = setInterval(() => {
```
Comment (piazzatron) [inline packages/web/src/pages/chat-page/components/ChatMessageList.tsx:63]: Intentional?

**Review by piazzatron — APPROVED** [review]
Little things:
- Z-Index issue w reactions
Screen Shot 2023-01-13 at 4 23 16 PM
- Looks like you can react to your own messages

**Comment by rickyrombo** [comment]
> Little things:
> 
> * Z-Index issue w reactions
> 
> Screen Shot 2023-01-13 at 4 23 16 PM

Will fix in future!

> - Looks like you can react to your own messages

I believe this is intended design. Julian has made a note on multiple reaction support

File: packages/web/src/components/data-entry/TextAreaV2.tsx (hunk 3 above)
Comment (rickyrombo) [reply packages/web/src/components/data-entry/TextAreaV2.tsx:71]: do you mean like rip it out or make standalone and compose in via props or something?

File: packages/web/src/pages/chat-page/ChatPage.tsx (hunk 4 above)
Comment (rickyrombo) [reply packages/web/src/pages/chat-page/ChatPage.tsx:57]: will think about it. it's the only place it's used now, so leaning towards no

File: packages/web/src/pages/chat-page/components/ChatComposer.tsx (hunk 5 above)
Comment (rickyrombo) [reply packages/web/src/pages/chat-page/components/ChatComposer.tsx:73]: added to doc

File: packages/web/src/pages/chat-page/ChatPage.tsx (hunk 4 above)
Comment (piazzatron) [reply packages/web/src/pages/chat-page/ChatPage.tsx:57]: 👍 imo nice to not have low level concerns like height measurements interspersed with higher level logic, DRY aside

File: packages/web/src/components/data-entry/TextAreaV2.tsx (hunk 3 above)
Comment (piazzatron) [reply packages/web/src/components/data-entry/TextAreaV2.tsx:71]: sorry just meant pull out the numbers into a const

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-composer

**Pull request merged**